src/workspace.rs

```diff
@@ -220,6 +220,31 @@ impl Workspace {
         Ok(())
     }
 
+    /// Remove all the contents of the caches in the workspace, freeing disk space.
+    pub fn purge_all_caches(&self) -> Result<(), Error> {
+        let mut paths = vec![
+            self.cache_dir(),
+            self.cargo_home().join("git"),
+            self.cargo_home().join("registry").join("src"),
+            self.cargo_home().join("registry").join("cache"),
+        ];
+
+        for index in std::fs::read_dir(self.cargo_home().join("registry").join("index"))? {
+            let index = index?;
+            if index.file_type()?.is_dir() {
+                paths.push(index.path().join(".cache"));
+            }
+        }
```

Comment:
Is there a reason to manually chain things instead of just running `remove_dir_all` on the top-level dirs?

Reply:
Yes, we want to keep the git repository of each index, but we want to remove the `.cache` directory inside each index.